docs/usage/CUSTOMER_SYNC.md

```diff
@@ -0,0 +1,106 @@
+and/or the [Type](https://docs.commercetools.com/api/projects/customers#set-custom-type) of their custom fields. 
+The `CustomerGroup` and `Type` references should be expanded with a key.
+Any reference that is not expanded will have its id in place and not replaced by the key will be considered as existing 
+resources on the target commercetools project and the library will issue an update/create an API request without reference
+resolution.
+
+ - When syncing from a source commercetools project, you can use [`mapToCustomerDrafts`](TODO: insert link once existing)
+    method that maps from a `Customer` to `CustomerDraft` to make them ready for reference resolution by the sync:
+
+    ````java
+    final List<CustomerDraft> customerDrafts = CustomerReferenceResolutionUtils.mapToCustomertDrafts(customerDrafts);
+    ````
+
+4. Create a `sphereClient` [as described here](IMPORTANT_USAGE_TIPS.md#sphereclient-creation).
+
+5. After the `sphereClient` is set up, a `CustomerSyncOptions` should be built as follows:
+````java
+// instantiating a CustomerSyncOptions
+final CustomerSyncOptions customerSyncOptions = CustomerSyncOptionsBuilder.of(sphereClient).build();
+````
+
+[More information about Sync Options](SYNC_OPTIONS.md).
+
+#### Running the sync
+When all prerequisites are fulfilled, follow those steps to run the sync:
+
+````java
+// instantiating a cart discount sync
+final CustomerSync customerSync = new CustomerSync(customerSyncOptions);
+
+// execute the sync on your list of customers
+CompletionStage<CustomerSyncStatistics> syncStatisticsStage = customerSync.sync(customerDrafts);
+````
+The result of the completing the `syncStatisticsStage` in the previous code snippet contains a `CustomerSyncStatistics`
+which contains all the stats of the sync process; which includes a report message, the total number of updated, created,
+failed, processed cart discounts, and the processing time of the last sync batch in different time units and in a
+human-readable format.
+
+````java
+final CustomerSyncStatistics stats = syncStatisticsStage.toCompletebleFuture().join();
+stats.getReportMessage();
+/*"Summary: 100 customers were processed in total (11 created, 87 updated, 2 failed to sync)."*/
+````
+
+__Note__ The statistics object contains the processing time of the last batch only. This is due to two reasons:
+
+ 1. The sync processing time should not take into account the time between supplying batches to the sync.
+ 2. It is not known by the sync which batch is going to be the last one supplied.
+
+#### More examples of how to use the sync
+
+ [Sync from an external source](https://github.com/commercetools/commercetools-sync-java/tree/master/src/integration-test/java/com/commercetools/sync/integration/externalsource/customers/CustomerSyncIT.java).
+
+*Make sure to read the [Important Usage Tips](IMPORTANT_USAGE_TIPS.md) for optimal performance.*
```

Comment:
Those are changed with the latest master, so you need to show example usages of the sync options.

Reply:
For example https://github.com/commercetools/commercetools-sync-java/blob/master/docs/usage/CART_DISCOUNT_SYNC.md#about-syncoptions

Reply:
I added the part about SyncOptions: d3048593.